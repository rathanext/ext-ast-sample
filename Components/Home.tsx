import {
  View,
  Text,
  StyleSheet,
  TextInput,
  Alert,
  ActivityIndicator,
} from "react-native";
import React from "react";
import { TouchableOpacity } from "react-native-gesture-handler";
// import Toast from "react-native-simple-toast";

const apikey = "xsGe8ZVLgT5UgRs03rioYt4P86EI0E4hvYC2h49y";

const Homepage = ({ navigation }: any) => {
  const [id, setId] = React.useState<string>("");
  const [loading, setLoading] = React.useState<boolean>(false);
  const [loadingran, setLoadingRan] = React.useState<boolean>(false);

  const getNasaData = async () => {
    setLoading(true);
    try {
      const response = await fetch(
        `https://api.nasa.gov/neo/rest/v1/neo/${id}?api_key=${apikey}`
      );

      console.log(response.status, "response");
      if (response?.status === 404) {
        alert("Enter correct id");
      } else if (response?.status === 200) {
        const json = await response.json();
        console.log(json, "apidata");
        navigation.navigate("Nasa", { data: json });
      } else {
        alert("something went wrong");
      }
      setLoading(false);
    } catch (err) {
      console.error(err);
      alert("some error" + JSON.stringify(err));
      setLoading(false);
    }
  };

  const getRandomId = async () => {
    setLoadingRan(true);
    try {
      const response = await fetch(
        `https://api.nasa.gov/neo/rest/v1/neo/browse?api_key=${apikey}`
      );
      const json = await response.json();
      console.log(
        json.near_earth_objects[Math.floor(Math.random() * 20)].id,
        "apidata random"
      );

      navigation.navigate("Nasa", {
        data: json.near_earth_objects[
          Math.floor(Math.random() * json.near_earth_objects.length)
        ],
      });

      setLoadingRan(false);
      //   if (json.id == id) {
      //     navigation.navigate("Nasa", { data: json });
      //   }
      //   setLoading(false);
    } catch (err) {
      console.error(err);
      setLoadingRan(false);
      //   Toast.show("Please enter correct ID", Toast.SHORT);
      //   setLoading(false);
    }
  };

  return (
    <View style={styles.container}>
      <View style={styles.idHolder}>
        <TextInput
          style={styles.input}
          onChangeText={setId}
          keyboardType={"numeric"}
          value={id}
          placeholder={"Enter Asteroid ID"}
          testID="asteroidid"
        />
      </View>
      <View style={styles.buttonHolder}>
        {!loading ? (
          <TouchableOpacity
            testID="submit"
            onPress={() => {
              getNasaData();
            }}
            disabled={id.length <= 0 ? true : false}
            style={[
              styles.button,
              { backgroundColor: id.length > 0 ? "blue" : "grey" },
            ]}
          >
            <Text testID="submitText" style={styles.buttonText}>
              Submit
            </Text>
          </TouchableOpacity>
        ) : (
          <View>
            <ActivityIndicator size={"small"} color={"red"} />
          </View>
        )}
        {!loadingran ? (
          <TouchableOpacity
            testID="random"
            onPress={() => {
              getRandomId();
            }}
            style={[
              styles.button,
              {
                backgroundColor: "blue",
                marginLeft: 10,
              },
            ]}
          >
            <Text testID="randomText" style={styles.buttonText}>
              Random
            </Text>
          </TouchableOpacity>
        ) : (
          <View>
            <ActivityIndicator size={"small"} color={"red"} />
          </View>
        )}
      </View>
    </View>
  );
};

export default Homepage;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  input: {
    width: "60%",
    height: 50,
    borderWidth: 1,
    paddingLeft: 10,
  },
  idHolder: {
    width: "100%",
    alignItems: "center",
  },
  buttonHolder: { marginTop: 10, flexDirection: "row", alignItems: "center" },
  button: {
    padding: 10,
    borderRadius: 5,
  },
  buttonText: {
    color: "#fff",
  },
});
