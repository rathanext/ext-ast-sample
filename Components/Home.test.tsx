import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";
import "@testing-library/jest-dom";

import Homepage from "./Home";
import Home from "./Homestack";

describe("CountryDetails Screen Test", () => {
  test("Crenders snapshot as expected", () => {
    const home = renderer.create(<Homepage />);
    let tree = home.toJSON();
    expect(tree).toMatchSnapshot();
  });
  // test("checking button", () => {
  //   const { getByTestId } = render(<Homepage />);
  //   expect(getByTestId("submit")).toBeCalled();
  // });
  // test("checking random button", () => {
  //   const { getByTestId } = render(<Homepage />);

  //   expect(getByTestId("submit")).toBeDisabled();
  // });

  test("check text content", () => {
    const { getByTestId } = render(<Homepage />);
    expect(getByTestId("randomText").children[0]).toBe("Random");
  });

  test("check text content", () => {
    const { getByTestId } = render(<Homepage />);
    expect(getByTestId("submitText").children[0]).toBe("Submit");
  });

  it("renders without crashing", () => {
    const home = renderer.create(<Homepage />);
    expect(home).toBeTruthy();
  });
  test("Check detail page run without crashes1", () => {
    const app = renderer.create(<Homepage />);
    const elementList = app.root.findAllByType("Text");
    expect(elementList.length).toBe(2);
  });

  test("Check detail page run without crashes1", () => {
    const app = renderer.create(<Homepage />);
    const elementList = app.root.findAllByType("View");
    expect(elementList.length).toBe(5);
  });

  test("Check detail page run without crashes1", () => {
    const app = renderer.create(<Homepage />);
    const elementList = app.root.findAllByType("TextInput");
    expect(elementList.length).toBe(1);
  });

  // test("Check Asteroid", () => {
  //   const { getByText, getAllByText } = render(<Homepage />);

  //   fireEvent.changeText(getByText("Enter Asteroid ID"), "2007234");
  //   fireEvent.press(getByText("Add the item to list"));

  //   const bananaElements = getAllByText("banana");
  //   expect(bananaElements).toHaveLength(1); // expect 'banana' to be on the list
  // });
});
