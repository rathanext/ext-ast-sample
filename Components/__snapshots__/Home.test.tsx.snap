// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetails Screen Test Crenders snapshot as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Enter Asteroid ID"
      rejectResponderTermination={true}
      style={
        Object {
          "borderWidth": 1,
          "height": 50,
          "paddingLeft": 10,
          "width": "60%",
        }
      }
      testID="asteroidid"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 10,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      exclusive={true}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      testID="submit"
    >
      <View
        accessible={true}
        style={
          Object {
            "backgroundColor": "grey",
            "borderRadius": 5,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
          testID="submitText"
        >
          Submit
        </Text>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      exclusive={true}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      testID="random"
    >
      <View
        accessible={true}
        style={
          Object {
            "backgroundColor": "blue",
            "borderRadius": 5,
            "marginLeft": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
          testID="randomText"
        >
          Random
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
